discovery/stackit/stackit.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stackit
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/url"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/version"
+
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/refresh"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+)
+
+const (
+	stackitLabelPrefix            = model.MetaLabelPrefix + "stackit_"
+	stackitLabelRole              = stackitLabelPrefix + "role"
+	stackitLabelProject           = stackitLabelPrefix + "project"
+	stackitLabelServerID          = stackitLabelPrefix + "server_id"
+	stackitLabelServerName        = stackitLabelPrefix + "server_name"
+	stackitLabelServerStatus      = stackitLabelPrefix + "server_status"
+	stackitLabelServerPowerStatus = stackitLabelPrefix + "server_power_status"
+	stackitLabelAvailabilityZone  = stackitLabelPrefix + "availability_zone"
```

Comment:
do we want all labels to contain the role `_server_` ? like in https://prometheus.io/docs/prometheus/latest/configuration/configuration/#instance-role (but some SD don't include _Role_ in)

Reply:
Yep, just for readability/consistency; if a label only makes sense for a role, it's better to have that in, I think (there is no rule AFAIK), the role itself is exposed as in a label, that should be enough for relabeling e.g.